### PR TITLE
Ci matrix for creating templates

### LINKED ISSUE #286: CI matrix for templates
As I'm changing each of the CI's for the models thoughts on doing a matrix and filters like we did in nf-core modules? More DRY, and it'll only get worse with more templates.

https://github.com/nf-core/modules/blob/master/.github/workflows/pytest-workflow.yml

### PR DESCRIPTION
**Pull request by Emiller88**
Closes #286 and 3. on #287

### CODE REVIEW

**Comment by Emiller88** [comment]
And now fixes #269 
But I figured those would fail because I knew that tensorflow and conda don't mix

**Comment by Zethson** [comment]
```
    In file included from src/cpp/cuda.cpp:4:0:
    src/cpp/cuda.hpp:14:10: fatal error: cuda.h: No such file or directory
     #include <cuda.h>
              ^~~~~~~~
    compilation terminated.
    error: command 'gcc' failed with exit status 1
```

Yeah, sadly was expecting this.

1. Our templates default to Docker by default and as long we as we don't have a nice mlf-core run command we can't switch that easily
2. The CUDA issue requires our additional CI which is still TBD

@Emiller88 Maybe remove Conda again? But the rest of your changes LGTM!

**Comment by Emiller88** [comment]
Should be good to go now that I've rebased!

File: .github/workflows/create_mlflow.yml [hunk 1]
```diff
@@ -1,12 +1,17 @@
-name: Create mlflow_tensorflow Template
+name: Create mlflow Template
 
 on: [push, pull_request]
 
 jobs:
-  build:
-
+  create:
```
Comment (Zethson) [inline .github/workflows/create_mlflow.yml:6]: +1 for renaming the jobs. Something I was particularly lazy it haha

**Review by Zethson — APPROVED** [review]
LGTM

**Pull request merged**